src/main.cpp

```diff
@@ -132,52 +154,132 @@ void DDWAF::dispose(const Napi::CallbackInfo& info) {
   return this->Finalize(info.Env());
 }
 
-void DDWAF::update(const Napi::CallbackInfo& info) {
-  mlog("calling update on DDWAF");
+Napi::Value DDWAF::update_config(const Napi::CallbackInfo& info) {
+  mlog("Calling update config on DDWAF");
 
   Napi::Env env = info.Env();
 
   if (this->_disposed) {
     Napi::Error::New(env, "Could not update a disposed WAF instance").ThrowAsJavaScriptException();
-    return;
+    return env.Undefined();
   }
 
-  if (info.Length() < 1) {
-    Napi::Error::New(env, "Wrong number of arguments, expected at least 1").ThrowAsJavaScriptException();
-    return;
+  if (info.Length() < 2) {
+    Napi::Error::New(env, "Wrong number of arguments, expected at least 2").ThrowAsJavaScriptException();
+    return env.Undefined();
   }
+
   if (!info[0].IsObject()) {
     Napi::TypeError::New(env, "First argument must be an object").ThrowAsJavaScriptException();
-    return;
+    return env.Undefined();
+  }
+
+  if (!info[1].IsString()) {
+    Napi::TypeError::New(env, "Second argument must be a string").ThrowAsJavaScriptException();
+    return env.Undefined();
   }
 
   ddwaf_object update;
-  mlog("building rules update");
+  mlog("Building config update");
   to_ddwaf_object(&update, env, info[0], 0, false, false, JsSet::Create(env), nullptr);
 
+  mlog("Obtaining config update path");
+  std::string config_path_str = info[1].As<Napi::String>().Utf8Value();
+  const char* config_path = config_path_str.c_str();
```

Comment:
at this point it might be worth putting this code in a function since it's used in a few places

Reply:
I've changed the code a little bit in order to simplify the way we get the char pointer to config_path.

Reply:
didn't change much, maybe we'll do some cleanup later, fine for now as is i guess